client/client-lib/src/lib.rs

```diff
@@ -629,7 +629,13 @@ impl Client<UserClientConfig> {
     pub async fn await_outgoing_contract_execution(&self, contract_id: ContractId) -> Result<()> {
         let gateway = self.fetch_active_gateway().await?;
         let future = reqwest::Client::new()
-            .post(&format!("{}pay_invoice", gateway.api))
+            .post(
+                gateway
+                    .api
+                    .join("pay_invoice")
+                    .expect("Could not join path to Url")
```

Comment:
Please change this to a `?`

Reply:
I assume since its better to use the library error in this case?